### ISSUE TITLE
Removing CancelledPacketHandleException

### ISSUE THREAD

**Issue by Mercenarioo** [TitaniumMC/Titanium#140]
I noticed that for every incoming packet CancelledPacketHandleException is being thrown. I think that hurts performance, are you able to remove it? I tried reporting it on other projects, but without results.

**Comment by P3ridot** [comment]
I checked the code and it would break a lot. Also it shouldn't hurt performance much

**Comment by Mercenarioo** [comment]
I don't think it would break a lot, it needs some changes for sure and throwing exception for every incoming packet is not what you want.

**Comment by MachineBreaker** [comment]
Unless you want to rework some of the PlayerConnectionUtil and NetworkManager logic then i don't recommend touching this critical code, without this, packets will be handled async even if you call the ensureMainThread method

**Comment by Mercenarioo** [comment]
That means the whole server might be slowed down badly, is there a better way to do this?

**Comment by MachineBreaker** [comment]
> It looks weird, I don't understand this too much. Packet is transformed to server thread then it's executed from the queue, right?

Some packets requires his logic to actually be executed on the Server's main thread due to no safer code, that's why it throws a exception to stop the execution and allows the NetworkManager to run it on the main thread

**Comment by Mercenarioo** [comment]
That's right, but it doesn't have anything to do with NetworkManager. The fix was simple and really easy, I don't know what improvement it brings, but removing this gives much cleaner, faster and more logic way of handling packets. I think it will save some CPU too.

**Comment by MachineBreaker** [comment]
> That's right, but it doesn't have anything to do with NetworkManager. The fix was simple and really easy, I don't know what improvement it brings, but removing this gives much cleaner, faster and more logic way of handling packets. I think it will save some CPU too.

NetworkManager is the one that catch this exception and run the packet in the proper thread

**Comment by MachineBreaker** [comment]
I don't consider this as a urgent patch due to his complexity and his low performance impact. I profiled the exception usage with YourKit on a 300on 1.8 server (Using my own spigot fork) and the performance impact is negligible

**Comment by MachineBreaker** [comment]
> How did you benchmark it?

With YourKit (?)
